contents/docker/Docker.md

```diff
@@ -0,0 +1,161 @@
+<frontmatter>
+  title: DevOps
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Docker
+
+Authors: [Rahul Rajesh](https://github.com/rrtheonlyone)
+
+### 1. What is Docker?
+
+Docker is a platform that is used to develop, deploy and run applications inside containers. This article will discuss some of the key ideas behind Docker, the reasons to use Docker and some resources to guide you in this process.
+
+#### 1.1 Virtual Machine (VM)
+
+Before the rise of Docker, people used VMs. A VM is an emulation of a real computer that executes program like a real computer. A VM is used to isolate an application and its dependancies into a self-contained unit that can run anywhere.
+
+![alt text](./Virtual_Machine.png "VM Diagram")
+
+With reference to the above diagram, VMs run on top of a physical machine using a “hypervisor”. A VM is able to provide virtual hardware, a kernel (i.e. OS) and user space. A Windows user can use a VM to run an application that is only supported on OSX for example. 
+
+#### 1.2 Container
+
+As mentioned above, Docker uses Containers, not VMs. The one big difference between containers and VMs is that containers *share* the host system’s kernel with other containers 
+
+![alt text](./Container.png "Container Diagram")
+
+This diagram shows you that containers package up just the user space, and not
+the kernel or virtual hardware like a VM does. Each container gets its own
+isolated user space to allow multiple containers to run on a single host
+machine. The operating system level architecture is being shared across containers. 
```

Comment:
After introducing container and VM, probably have a small section telling user explicitly what are the benefits of container over VMs(More efficient space usage, less performance overhead?)